main.ts

```diff
@@ -85,9 +82,81 @@ class SettingTab extends PluginSettingTab {
 					.setPlaceholder("Enter your API key")
 					.setValue(this.plugin.settings.openAIApiKey)
 					.onChange(async (value) => {
-						this.plugin.settings.openAIApiKey = value;
-						await this.plugin.saveSettings();
+						await this.plugin.updateSettings({openAIApiKey: value});
+					}),
+			);
+		new Setting(containerEl)
+			.setName("Minimum Summary Size")
+			.setDesc("Minimum number key points to include in generated summary.")
+			.addText((text) =>
+				text
+					.setPlaceholder("3")
+					.setValue("" + this.plugin.settings.summarySize)
+					.onChange(async (value) => {
+						let parsed = parseInt(value);
+						if (isNaN(parsed)) return;
+						parsed = parsed > MAX_SUMMARY_SIZE ? MAX_SUMMARY_SIZE : 
+								 parsed < DEFAULT_SUMMARY_SIZE ? DEFAULT_SUMMARY_SIZE : parsed;
```

Comment:
Reference for this parsing logic: https://github.com/blacksmithgu/obsidian-dataview/blob/master/src/main.ts#L479